[TLVB-92] 마켓 생성, 조회 API 구현

nasaoreo opened the pull request:
## 👩‍💻 요구 사항과 구현 내용
- 마켓 생성 API 구현
- 마켓 조회 API 구현

## ✅ 피드백 반영사항

## 🤔 PR 포인트 & 궁금한 점
- Entity에 대한 검증은 Entity 검증 오류에 대한 예외 처리 Convention을 정한 뒤 작업할 예정입니다.
- 리뷰, 즐겨찾기, 좋아요 도메인 기능이 구현되면, 마켓에 대한 리뷰, 즐겨찾기, 좋아요를 얻어오는 로직을 추가할 예정입니다.
- 컨트롤러 함수에서 User 엔티티를 바로 얻어오는 부분은 @heoseungyeon 의 작업이 완료되면 UserDetail 타입으로 변경할 예정입니다.
- Controller에 대한 테스트는 멘토님이 알려주신 `@WebMvcTest`를 사용하려고 하였지만, Security 관련한 빈 생성 오류가 나네요.. 이 부분은 더 조사한 뒤 변경할 예정입니다.

## 관련 이슈 
TLVB-92

nasaoreo committed:
fix: SignUpRequest의 생성자 추가

- Lombok @NoArgsConstructor 어노테이션 사용

nasaoreo committed:
feat: 마켓 생성, 조회 기능 추가

Heejeong01110 commented on src/main/java/kdt/prgrms/kazedon/everevent/controller/MarketController.java:
```diff
@@ -0,0 +1,43 @@
+    @PostMapping("markets")
+    public ResponseEntity createMarket(@Valid @RequestBody MarketCreateRequest marketCreateRequest,
+                                       @AuthUser User user){
+        marketService.createMarket(marketCreateRequest, user);
```
이부분도 userId를 전달하면 좋을 것 같아요~!!

nasaoreo replied:
> 이부분도 userId를 전달하면 좋을 것 같아요~!!

이 부분은 에디가 작업 끝나면 일괄적으로 하신대서, 그 이후에 작업할 예정입니당!

nasaoreo replied:
> 이부분도 userId를 전달하면 좋을 것 같아요~!!

그러네용!!ㅎㅎ 수정하겠습니다~!

Heejeong01110 commented on src/main/java/kdt/prgrms/kazedon/everevent/controller/MarketController.java:
```diff
@@ -0,0 +1,43 @@
+                                       @AuthUser User user){
+        marketService.createMarket(marketCreateRequest, user);
+
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .build()
+                .toUri();
```
이부분은 어떤 주소를 생성하나욤??

nasaoreo replied:
> 이부분은 어떤 주소를 생성하나욤??

Market 목록을 조회하는 URI로 Location을 지정하였습니다!

Heejeong01110 commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/market/dto/SimpleMarket.java:
```diff
@@ -0,0 +1,14 @@
+@Getter
+@Builder
+public class SimpleMarket {
+    long marketId;
+    String description;
+    int eventCount;
+    int favoriteCount;
+    int reviewCount;
```
접근지정자가 빠진 것 같아요!!

nasaoreo replied:
> 접근지정자가 빠진 것 같아요!!

어머! 감사합니당~! 수정할게용!!

Heejeong01110 reviewed (approved):
시큐리티쪽에서 계속 문제가 생겼을텐데 구현 잘 하신 것 같아서 다행이네요👍👍 고생하셨습니다~!

nasaoreo committed:
refactor: 가게 생성 코드 리팩토링

- EventService의 가게 생성 시 userId 사용하도록 변경
- SimpleMarket DTO 멤버 변수의 한정자 추가
- 코드 변경에 따른 테스트 코드 수정

Heejeong01110 commented on src/main/java/kdt/prgrms/kazedon/everevent/service/MarketService.java:
```diff
@@ -0,0 +1,42 @@
+    private final MarketConverter marketConverter;
+    private final UserRepository userRepository;
+
+    public MarketReadResponse getMarketsByUser(Long userId, Pageable pageable){
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.USER_NOT_FOUNDED, userId));
+
+        Page<SimpleMarket> simpleMarkets = marketRepository
+                .findByUser(user, pageable)
+                .map(market -> marketConverter.convertToSimpleMarket(market, 0, 0));
+
+        return marketConverter.convertToMarketReadResponse(simpleMarkets);
+    }
```
이부분에 @transactional(readOnly = true) 옵션 주면 좋을 것 같아요~!

nasaoreo replied:
> 이부분에 @transactional(readOnly = true) 옵션 주면 좋을 것 같아요~!

오 너무 좋은 팁 감사해요! 이렇게 잘 붙이지 않았는데, 성능 향상에 큰 도움이 되겠네요! 
지금 이벤트 기능도 작업하고 있는데, 일괄적으로 적용해서 다음 PR에 반영할게요~!

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/market/repository/MarketRepository.java:
```diff
@@ -1,7 +1,16 @@
 
+@Repository
 public interface MarketRepository extends JpaRepository<Market, Long> {
+    @Query("select m from Market m where m.user = :user")
```
쿼리 예약어는 대문자로 작성해서 구분하면 좋을거 같네요.

ksy90101 replied:
근데 이걸 굳이 쿼리로 작성해야 하나요?

nasaoreo replied:
맞습니다, 쿼리문으로 작성하지 않아도 되네요! 제거 할게요~!

nasaoreo replied:
> 근데 이걸 굳이 쿼리로 작성해야 하나요?

맞습니다, 쿼리문으로 작성하지 않아도 되네요! 제거 할게요~!

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/exception/GlobalExceptionHandler.java:
```diff
@@ -1,7 +1,23 @@
 public class GlobalExceptionHandler {
+    @ExceptionHandler
+    public ResponseEntity<String> handleNotFound(NotFoundException exception) {
+        log.warn(exception.getMessage());
```
예외가 발생한건데 warn 레벨이 맞을까요?

nasaoreo replied:
> 예외가 발생한건데 warn 레벨이 맞을까요?

이 부분을 신경쓰지 못했어요! error로 변경하겠습니다!

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/service/MarketService.java:
```diff
@@ -0,0 +1,45 @@
+
+        Page<SimpleMarket> simpleMarkets = marketRepository
+                .findByUser(user, pageable)
+                .map(market -> marketConverter.convertToSimpleMarket(market, 0, 0));
```
0의 의미는 무엇인가요?
단순히 봤을땐 의미를 찾기가 힘들어서 상수로 해주시면 어떨까요?
이런걸 매직넘버라고 합니다 ㅎㅎ

nasaoreo replied:
> 0의 의미는 무엇인가요? 단순히 봤을땐 의미를 찾기가 힘들어서 상수로 해주시면 어떨까요? 이런걸 매직넘버라고 합니다 ㅎㅎ

이것은 가게의 좋아요, 리뷰 수를 계산한 뒤 그 값을 설정하는 부분인데 아직 해당 도메인의 기능이 구현되지 않아 default 값으로 설정하였습니다. 이 부분을 변수로 만들어 대체해야겠네요! 조언 감사합니당!

nasaoreo committed:
refactor: 로깅 레벨 수정, 상수 값 변수화, 불필요한 쿼리문 삭제

dia312 reviewed (approved):
진아님 고생하셨씁니다!!! dto 레벨에서 예외처리하는 부분 많이 참고했습니당 ㅎㅎ 👍
